engine/src/test/java/io/zeebe/engine/util/JobClient.java

```diff
@@ -17,25 +17,65 @@
  */
 package io.zeebe.engine.util;
 
-import static io.zeebe.protocol.intent.JobIntent.COMPLETED;
-import static io.zeebe.protocol.intent.JobIntent.FAILED;
-import static io.zeebe.protocol.intent.JobIntent.RETRIES_UPDATED;
-
 import io.zeebe.exporter.api.record.Record;
 import io.zeebe.exporter.api.record.value.JobRecordValue;
 import io.zeebe.protocol.impl.encoding.MsgPackConverter;
 import io.zeebe.protocol.impl.record.value.job.JobRecord;
 import io.zeebe.protocol.intent.JobIntent;
 import io.zeebe.test.util.record.RecordingExporter;
+import java.util.function.Function;
 import org.agrona.DirectBuffer;
 import org.agrona.concurrent.UnsafeBuffer;
 
 public class JobClient {
+  private static final Function<Long, Record<JobRecordValue>> COMPLETE_SUCCESS_SUPPLIER =
+      (position) ->
+          RecordingExporter.jobRecords(JobIntent.COMPLETED)
+              .withSourceRecordPosition(position)
+              .getFirst();
+
+  private static final Function<Long, Record<JobRecordValue>> COMPLETE_REJECTION_SUPPLIER =
+      (position) ->
+          RecordingExporter.jobRecords(JobIntent.COMPLETE)
+              .onlyCommandRejections()
+              .withSourceRecordPosition(position)
+              .getFirst();
+
+  private static final Function<Long, Record<JobRecordValue>> FAIL_SUCCESS_SUPPLIER =
+      (position) ->
+          RecordingExporter.jobRecords(JobIntent.FAILED)
+              .withSourceRecordPosition(position)
+              .getFirst();
+
+  private static final Function<Long, Record<JobRecordValue>> FAIL_REJECTION_SUPPLIER =
+      (position) ->
+          RecordingExporter.jobRecords(JobIntent.FAIL)
+              .onlyCommandRejections()
+              .withSourceRecordPosition(position)
+              .getFirst();
+
+  private static final Function<Long, Record<JobRecordValue>> UPDATE_RETRIES_SUCCESS_SUPPLIER =
+      (position) ->
+          RecordingExporter.jobRecords(JobIntent.RETRIES_UPDATED)
+              .withSourceRecordPosition(position)
+              .getFirst();
+
+  private static final Function<Long, Record<JobRecordValue>> UPDATE_RETRIES_REJECTION_SUPPLIER =
```

Comment:
I think they can be summarized as:

```
private static final Function<Long, Record<JobRecordValue>> SUCCESS_SUPPLIER =
      (position) ->
          RecordingExporter.jobRecords()
              .onlyEvents()
              .withSourceRecordPosition(position)
              .getFirst();

  private static final Function<Long, Record<JobRecordValue>> REJECTION_SUPPLIER =
      (position) ->
          RecordingExporter.jobRecords()
              .onlyCommandRejections()
              .withSourceRecordPosition(position)
              .getFirst();

```

Then you do not need to distinguish between them